src/main/java/net/fabricmc/loom/util/MojmapCreator.java

```diff
@@ -0,0 +1,244 @@
+import net.fabricmc.mapping.tree.TinyMappingFactory;
+
+public class MojmapCreator {
+	private static final Gson GSON = new Gson();
+
+	public static Dependency mojmap(Project project, LoomGradleExtension extension, String minecraftVersionRaw) throws IOException {
+		File userCache = extension.getUserCache();
+		File mappingsDir = new File(userCache, "mappings");
+		File steps = new File(mappingsDir, "steps");
+
+		ManifestVersion.Versions version = MinecraftProvider.findVersion(project, extension, minecraftVersionRaw).get();
+		String id = version.id;
+
+		File mappingsJar = new File(mappingsDir, "mojmap-" + id + ".jar");
+
+		if (!mappingsJar.exists() || project.getGradle().getStartParameter().isRefreshDependencies()) {
+			MappingSet officialToNamed;
+			MappingSet intermediaryToOfficial;
+
+			// Download and parse Proguard logs
+			{
+				File minecraftJson = new File(userCache, "minecraft-" + id + "-info.json");
+				DownloadUtil.downloadIfChanged(new URL(version.url), minecraftJson, project.getLogger());
+
+				MinecraftVersionInfo info = GSON.fromJson(MoreFiles.asCharSource(minecraftJson.toPath(), StandardCharsets.UTF_8).read(), MinecraftVersionInfo.class);
+
+				File clientMappings = new File(steps, "mojmap-" + id + "-client.txt");
+				File serverMappings = new File(steps, "mojmap-" + id + "-server.txt");
+				String clientMappingsUrl = info.downloads.get("client_mappings").url;
+				String serverMappingsUrl = info.downloads.get("server_mappings").url;
+
+				DownloadUtil.downloadIfChanged(new URL(clientMappingsUrl), clientMappings, project.getLogger());
+				DownloadUtil.downloadIfChanged(new URL(serverMappingsUrl), serverMappings, project.getLogger());
+
+				MappingSet mappings = MappingSet.create();
+
+				// Print license headers
+				String clientMappingsString = MoreFiles.asCharSource(clientMappings.toPath(), StandardCharsets.UTF_8).read();
+				String serverMappingsString = MoreFiles.asCharSource(serverMappings.toPath(), StandardCharsets.UTF_8).read();
+
+				{
+					project.getLogger().warn("USING OFFICIAL MAPPINGS. YOU ARE USING THESE MAPPINGS AT YOUR OWN RISK UNDER THE FOLLOWING LICENSE:");
+					project.getLogger().warn("");
+
+					for (String line : clientMappingsString.split("\\v+")) {
+						if (line.startsWith("#")) {
+							project.getLogger().warn(line);
+						}
+					}
+
+					project.getLogger().warn("");
+				}
+
+				try (
+						ProGuardReader clientReader = new ProGuardReader(new StringReader(clientMappingsString));
+						ProGuardReader serverReader = new ProGuardReader(new StringReader(serverMappingsString))) {
+					clientReader.read(mappings);
+					serverReader.read(mappings);
+				}
+
+				officialToNamed = mappings.reverse();
+			}
+
+			// Download and parse intermediaries
+			{
+				File intermediaryJar = new File(steps, "v2-intermediary-" + id + ".jar");
+				DownloadUtil.downloadIfChanged(new URL(extension.getIntermediaryUrl().apply(id)), intermediaryJar, project.getLogger());
```

Comment:
This is done else where, just use what has already been done. Depends on moving this logic to a later time.